app/src/main/java/nu/yona/app/YonaApplication.java

```diff
@@ -202,5 +209,53 @@ public static Tracker getTracker()
 		}
 		return mContext.tracker;
 	}
+
+	protected OpenVPNService mService;
+
+	private final ServiceConnection mConnection = new ServiceConnection()
+	{
+
+		@Override
+		public void onServiceConnected(ComponentName className,
+									   IBinder service)
+		{
+			// We've bound to LocalService, cast the IBinder and get LocalService instance
+			OpenVPNService.LocalBinder binder = (OpenVPNService.LocalBinder) service;
+			mService = binder.getService();
+			stopVPN();
+			getAppContext().unbindService(mConnection);
+		}
+
+		@Override
+		public void onServiceDisconnected(ComponentName arg0)
+		{
+			mService = null;
+			// TODO: Need to show user a alert on failure to stop vpn.
+		}
+
+	};
+
+
+	private void stopVPN()
+	{
+		ProfileManager.setConntectedVpnProfileDisconnected(YonaActivity.getActivity());
+		if (mService != null && mService.getManagement() != null)
+		{
+			mService.getManagement().stopVPN(false);
+		}
+	}
+
+	private void bindOpenVPNService()
+	{
+		Intent intent = new Intent(YonaActivity.getActivity(), OpenVPNService.class);
+		intent.setAction(OpenVPNService.START_SERVICE);
+		getAppContext().bindService(intent, mConnection, Context.BIND_AUTO_CREATE);
+	}
+
+	public void stopOpenVPNService()
+	{
+		bindOpenVPNService();
```

Comment:
Inline that method here, as it is pointless on its own.

Reply:
bindOpenVPNService can be used in future, to do any other operations on service other than stopping.

Reply:
That's not the case. This method binds a handler that stops the VPN, so it should be named ``stopOpenVPNService``.